p2p/conn/connection.go

```diff
@@ -200,15 +202,56 @@ func (c *MConnection) OnStart() error {
 	return nil
 }
 
-// OnStop implements BaseService
-func (c *MConnection) OnStop() {
+// FlushStop replicates the logic of OnStop,
+// but flushes all pending msgs before closing the connection.
+func (c *MConnection) FlushStop() {
```

Comment:
Is this safe with concurrent/repeated FlushStop calls?  The Service Stop/OnStop mechanism ensures that OnStop only gets called once.  Not so here.